### PR TITLE
Python module version export fixes

### LINKED ISSUE #51: Version exporting of `leidenalg` and `SpatialDE` does not work
### Description of the bug

Upon checking the contents of `pipeline_info/versions.yml` I found that the entries for `leidenalg` and `SpatialDE` were empty, which they should not be. It seems that using `<module>.__version__` does not work for these modules, for some reason. Tested with both Conda and the Docker image utilised in the pipeline. Partial fix in #50.

### Command used and terminal output

```console
$ python -c "import SpatialDE; print(SpatialDE.__version__)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: module 'SpatialDE' has no attribute '__version__'
```

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by fasterius**
This PR fixes the version exporting of the `leidenalg` Python module, but more work is needed for the `SpatialDE` module. The change for `leidenalg` was simply to go from `leidenalg.__version__` to `leidenalg.version`, but no such simple change works for `SpatialDE`. Installing `setuptools` and various other possible solutions from Google did not work either.

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [X] Make sure your code lints (`nf-core lint`).
- [X] Fix `SpatialDE` version export.
- [X] `CHANGELOG.md` is updated.

### CODE REVIEW

**Comment by grst** [comment]
Did you try `importlib.metadata`? 
https://docs.python.org/3/library/importlib.metadata.html#overview

**Comment by fasterius** [comment]
Awesome, that worked!

File: modules/local/st_spatial_de.nf [hunk 1]
```diff
@@ -44,9 +45,9 @@ process ST_SPATIAL_DE {
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":
         quarto: \$(quarto -v)
-        leidenalg: \$(python -c "import leidenalg; print(leidenalg.__version__)")
+        leidenalg: \$(python -c "import leidenalg; print(leidenalg.version)")
         scanpy: \$(python -c "import scanpy; print(scanpy.__version__)")
-        SpatialDE: \$(python -c "import SpatialDE; print(SpatialDE.__version__)")
+        SpatialDE: \$(python -c "import SpatialDE; from importlib.metadata import version; print(version('SpatialDE'))")
```
Comment (grst) [inline modules/local/st_spatial_de.nf]: the import here shouldn't even be necessary. 

```suggestion
        SpatialDE: \$(python -c "from importlib.metadata import version; print(version('SpatialDE'))")
```

**Review by grst — APPROVED** [review]

**Comment by fasterius** [comment]
Hmm, I can't seem to run the `nf-test` tests locally now to update the snapshots, and the test data seems to point to non-existent local files. Was this something that was supposed to change by pointing to the testdata repo, or am I doing something wrong, @grst ?

**Comment by grst** [comment]
You'll need to download the test data manually to run tests locally by executing
```
git clone --depth 1 --branch spatialtranscriptomics https://github.com/nf-core/test-datasets.git
``` 
in the root dir of the pipeline. 

Before, samplesheets were using `https` links directly. With folders this is not possible anymore. 

Not sure if this could be automated as part of the nf-tests. At the very least we should document this somewhere.

**Pull request merged**